File: lua/bob/bob.lua
local bob = {}

local string = string

local responses = {
    ["statement"] = "Whatever",
    ["question"] = "Sure",
    ["yelling"] = "Whoa, chill out!",
    ["empty"] = "Fine, be that way."
}

function string.ends(str, endswith)
    return endswith == '' or string.sub(str, -string.len(endswith)) == endswith
end

function bob.hey(str)
    if string.len(str) == 0 then
        return responses["empty"]
    elseif string.ends(str, "?") then
        return responses["question"]
    elseif str == string.upper(str) then
        return responses["yelling"]
    else
        return responses["statement"]
    end
end

return bob

File: ruby/bob/bob.rb
class Bob
  class << self
    def hey(phrase)
      phrase.gsub!(/\s+/, '')

      return 'Fine. Be that way!' if phrase.strip.empty?

      if phrase.end_with?('?')
        return 'Sure.'                              if phrase.include?(':)')
        return 'Calm down, I know what I\'m doing!' if phrase.upcase.gsub(/\d+/, '') == phrase
        return 'Sure.'
      end

      return 'Whoa, chill out!' if phrase.upcase == phrase && phrase.downcase != phrase
      'Whatever.'
    end
  end
end

module BookKeeping
  VERSION = 2
end
